### PR TITLE
qpdf: new package

### PR DESCRIPTION
**Pull request by taliaferro**
Added the package `qpdf` which provides a library and CLI tool to combine, linearize, transform and encrypt PDF documents. It has variants for different cryptography providers (see "[build-time crypto providers](https://qpdf.readthedocs.io/en/latest/installation.html#build-time-crypto-selection)" in the manual for details.)

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed homepage and version sha256.

File: var/spack/repos/builtin/packages/qpdf/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on("gnutls", when="crypto=gnutls")
+
+    def cmake_args(self):
+
+        if self.spec.satisfies("crypto=implicit"):
+            return []
+
+        else:
+            crypto_type = self.spec.variants["crypto"].value.upper()
+            return ["USE_IMPLICIT_CRYPTO=0", f"REQUIRE_CRYPTO_{crypto_type}=1"]
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/qpdf/package.py]:
```suggestion
        args = []
        if not self.spec.satisfies("crypto=implicit"):
            crypto_type = self.spec.variants["crypto"].value.upper()
            args.append("USE_IMPLICIT_CRYPTO=0")
            args.append(f"REQUIRE_CRYPTO_{crypto_type}=1")
        return args
```

**Review by alecbcs — COMMENTED** [review]
Not required, but here's an optional suggestion to clean up the args section a bit.

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks for the addition @taliaferro

**Pull request merged**